### PR TITLE
fix msvc build

### PR DESCRIPTION
**Pull request by jef-n**

### CODE REVIEW

**Comment by jef-n** [comment]
https://cdash.orfeo-toolbox.org/viewBuildError.php?buildid=95641

File: src/core/mesh/qgsmeshlayerprofilegenerator.h [hunk 1]
```diff
@@ -24,14 +24,14 @@
 #include "qgscoordinatetransformcontext.h"
 #include "qgscoordinatetransform.h"
 #include "qgstriangularmesh.h"
```
Comment (nyalldawson) [inline src/core/mesh/qgsmeshlayerprofilegenerator.h:26]: Does adding an explicit destructor avoid this?

Comment (jef-n) [reply src/core/mesh/qgsmeshlayerprofilegenerator.h:26]: > Does adding an explicit destructor avoid this?

Like that?  No.

```diff
diff --git a/src/core/mesh/qgsmeshlayerprofilegenerator.cpp b/src/core/mesh/qgsmeshlayerprofilegenerator.cpp
index 206c630ba2..f4c5c07f7e 100644
--- a/src/core/mesh/qgsmeshlayerprofilegenerator.cpp
+++ b/src/core/mesh/qgsmeshlayerprofilegenerator.cpp
@@ -26,8 +26,10 @@
 #include "qgsmeshlayerelevationproperties.h"
 
 //
-// QgsMeshLayerProfileGenerator
+// QgsMeshLayerProfileResults
+//
 //
+QgsMeshLayerProfileResults::~QgsMeshLayerProfileResults() = default;
 
 QString QgsMeshLayerProfileResults::type() const
 {
diff --git a/src/core/mesh/qgsmeshlayerprofilegenerator.h b/src/core/mesh/qgsmeshlayerprofilegenerator.h
index 9a3119a556..6927dd61a7 100644
--- a/src/core/mesh/qgsmeshlayerprofilegenerator.h
+++ b/src/core/mesh/qgsmeshlayerprofilegenerator.h
@@ -47,6 +47,7 @@ class CORE_EXPORT QgsMeshLayerProfileResults : public QgsAbstractProfileResults
 {
 
   public:
+    ~QgsMeshLayerProfileResults() override;
 
     QgsPointSequence rawPoints;
     QMap< double, double> results;
diff --git a/src/core/plot/qgsplot.cpp b/src/core/plot/qgsplot.cpp
index c05852828b..eab4c3adf6 100644
--- a/src/core/plot/qgsplot.cpp
+++ b/src/core/plot/qgsplot.cpp
@@ -448,10 +448,10 @@ void Qgs2DPlot::calculateOptimisedIntervals( QgsRenderContext &context )
 
   QgsNumericFormatContext numericContext;
 
-  auto refineIntervalForAxis = []( double axisMinimum, double axisMaximum,
-                                   const std::function< double( double ) > &sizeForLabel,
-                                   double availableSize, double idealSizePercent, double sizeTolerancePercent,
-                                   double & labelInterval, double & majorInterval, double & minorInterval )
+  auto refineIntervalForAxis = [&]( double axisMinimum, double axisMaximum,
+                                    const std::function< double( double ) > &sizeForLabel,
+                                    double availableSize, double idealSizePercent, double sizeTolerancePercent,
+                                    double & labelInterval, double & majorInterval, double & minorInterval )
   {
     auto roundBase10 = []( double value )->double
     {
diff --git a/src/core/raster/qgsrasterlayerprofilegenerator.cpp b/src/core/raster/qgsrasterlayerprofilegenerator.cpp
index 16e2baaa35..fe77b64593 100644
--- a/src/core/raster/qgsrasterlayerprofilegenerator.cpp
+++ b/src/core/raster/qgsrasterlayerprofilegenerator.cpp
@@ -29,6 +29,7 @@
 //
 // QgsRasterLayerProfileResults
 //
+QgsRasterLayerProfileResults::~QgsRasterLayerProfileResults() = default;
 
 QString QgsRasterLayerProfileResults::type() const
 {
diff --git a/src/core/raster/qgsrasterlayerprofilegenerator.h b/src/core/raster/qgsrasterlayerprofilegenerator.h
index 8a1a290c32..b556d2ba89 100644
--- a/src/core/raster/qgsrasterlayerprofilegenerator.h
+++ b/src/core/raster/qgsrasterlayerprofilegenerator.h
@@ -45,6 +45,7 @@ class CORE_EXPORT QgsRasterLayerProfileResults : public QgsAbstractProfileResult
 {
 
   public:
+    ~QgsRasterLayerProfileResults() override;
 
     QgsPointSequence rawPoints;
     QMap< double, double > results;
diff --git a/src/core/vector/qgsvectorlayerprofilegenerator.cpp b/src/core/vector/qgsvectorlayerprofilegenerator.cpp
index 0895a1f022..e1a3cb3913 100644
--- a/src/core/vector/qgsvectorlayerprofilegenerator.cpp
+++ b/src/core/vector/qgsvectorlayerprofilegenerator.cpp
@@ -38,6 +38,8 @@
 // QgsVectorLayerProfileResults
 //
 
+QgsVectorLayerProfileResults::~QgsVectorLayerProfileResults() = default;
+
 QString QgsVectorLayerProfileResults::type() const
 {
   return QStringLiteral( "vector" );
diff --git a/src/core/vector/qgsvectorlayerprofilegenerator.h b/src/core/vector/qgsvectorlayerprofilegenerator.h
index c563c58934..9889282b63 100644
--- a/src/core/vector/qgsvectorlayerprofilegenerator.h
+++ b/src/core/vector/qgsvectorlayerprofilegenerator.h
@@ -51,6 +51,7 @@ class QgsMarkerSymbol;
 class CORE_EXPORT QgsVectorLayerProfileResults : public QgsAbstractProfileResults
 {
   public:
+    ~QgsVectorLayerProfileResults() override;
 
     QgsPointSequence rawPoints;
     QMap< double, double > mDistanceToHeightMap;
```

Comment (nyalldawson) [reply src/core/mesh/qgsmeshlayerprofilegenerator.h:26]: ok, thanks for the test!

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**